agent/engine/interface.go

```diff
@@ -43,6 +43,9 @@ type TaskEngine interface {
 	// lifecycle. If it returns an error, the task was not added.
 	AddTask(*apitask.Task)
 
+	// UpdateTask updates a task in the task engine.
```

Comment:
From the implementation it looks like it also adds the task to the task engine. We might want to make the description more precise for future readers.

Reply:
Will mirror the changes mentioned here: https://github.com/aws/amazon-ecs-agent/pull/4301#discussion_r1729599638 to this comment as well, which should address this.